client/app/lib/changeteam/itemview.coffee

```diff
@@ -0,0 +1,65 @@
+kd      = require 'kd'
+JView   = require 'app/jview'
+globals = require 'globals'
+
+module.exports = class ChangeTeamListItem extends kd.ListItemView
+
+  JView.mixin @prototype
+
+  constructor: (options = {}, data) ->
+
+    options.cssClass = kd.utils.curry 'change-team-item', options.cssClass
+
+    super options, data
+
+    team = @getData()
+    { customize, slug } = team
+
+    @logoWrapper = new kd.CustomHTMLView
+      cssClass : 'team-logo-wrapper'
+    if logo = team.customize?.logo
+      @logoWrapper.addSubView new kd.CustomHTMLView
+        tagName    : 'img'
+        attributes :
+          src      : logo
+    else
+      @logoWrapper.addSubView new kd.CustomHTMLView
+        cssClass   : 'default-team-logo'
+
+    { groupsController } = kd.singletons
+    currentGroup = groupsController.getCurrentGroup()
+
+    if currentGroup.slug is slug
+      @actionElement = new kd.CustomHTMLView
+        tagName  : 'span'
+        cssClass : 'current-team-label'
+        partial  : 'Current Team'
+      return
+
+    hostname     = globals.config.domains.main
+    domain       = if slug is 'koding' then hostname else "#{slug}.#{hostname}"
+
+    hasInvitation  = team.invitationCode?
+    actionTitle    = if hasInvitation then 'Join' else 'Switch'
+    actionLink     = if hasInvitation
+    then "//#{domain}/Invitation/#{encodeURIComponent team.invitationCode}"
+    else "//#{domain}/-/loginwithtoken?token=#{team.jwtToken}"
```

Comment:
it is very nice solution to use JWT here since the `loginwithtoken` endpoint does all the required heavy lifting on logging someone in, but just be aware that tokens has a validity duration, after the duration they wont let someone in. Say if someone left koding in a tab for 1 week and when they come back, if they try to switch teams they wont be able to login